test-data/unit/check-possibly-undefined.test

```diff
@@ -1043,3 +1043,126 @@ def foo(x: Union[int, str]) -> None:
+# was incorrectly reported as undefined when except has continue
+def random() -> float: return 0.5
+
+if random():
+    for i in range(10):
+        try:
+            value = random()
+            break
+        except Exception:
+            continue
+    else:
+        raise RuntimeError
+
+    print(value)  # Should not error - value is defined if we broke
+else:
+    value = random()
+
+print(value)  # Should not error
+[builtins fixtures/for_else_exception.pyi]
+[typing fixtures/typing-medium.pyi]
+
+[case testForElseWithBreakInTryExceptContinueNoIf]
+# flags: --enable-error-code possibly-undefined
+# Simpler version without if statement
+def random() -> float: return 0.5
+
+for i in range(10):
+    try:
+        value = random()
+        break
+    except Exception:
+        continue
+else:
+    raise RuntimeError
+
+print(value)  # Should not error
+[builtins fixtures/for_else_exception.pyi]
+[typing fixtures/typing-medium.pyi]
+
+[case testForElseWithBreakInTryExceptPass]
+# flags: --enable-error-code possibly-undefined
+# Version with pass instead of continue - should also work
+def random() -> float: return 0.5
+
+if random():
+    for i in range(10):
+        try:
+            value = random()
+            break
+        except Exception:
+            pass
+    else:
+        raise RuntimeError
+
+    print(value)  # Should not error
+else:
+    value = random()
+
+print(value)  # Should not error
+[builtins fixtures/for_else_exception.pyi]
+[typing fixtures/typing-medium.pyi]
+
+[case testForElseWithConditionalDefBeforeBreak]
+# flags: --enable-error-code possibly-undefined
+# Test that conditional definition before break still works correctly
+def random() -> float: return 0.5
+
+if random():
+    for i in range(10):
+        try:
+            if i > 10:
+                value = random()
+                break
+        except Exception:
+            continue
+    else:
+        raise RuntimeError
+
+    print(value)  # Should not error (though might be undefined at runtime)
+else:
+    value = random()
+
+print(value)  # Should not error
+[builtins fixtures/for_else_exception.pyi]
+[typing fixtures/typing-medium.pyi]
+
+[case testForElseDefineInBothBranches]
+# flags: --enable-error-code possibly-undefined
+# Test that variable defined in both for break and else branches is not undefined
+for i in range(10):
+    if i:
+        value = i
+        break
+else:
+    value = 1
+
+print(value)  # Should not error - value is defined in all paths
+[builtins fixtures/for_else_exception.pyi]
+[typing fixtures/typing-medium.pyi]
+
+[case testForElseWithWalrusInBreak]
+# flags: --enable-error-code possibly-undefined
+# Test with walrus operator in if condition before break
+def random() -> float: return 0.5
+
+if random():
+    for i in range(10):
+        if value := random():
+            break
+    else:
+        raise RuntimeError
+
+    print(value)  # Should not error - value is defined if we broke
```

Comment:
Maybe also add one two tests where it should error? Like, e.g, if `raise` above is conditional.

Reply:
Good idea, I will add the test.